box_embeddings/modules/tf_embedding.py

```diff
@@ -0,0 +1,72 @@
+from typing import List, Tuple, Union, Dict, Any, Optional
+import tensorflow as tf
+from box_embeddings.parameterizations.tf_box_tensor import (
+    TFBoxFactory,
+    TFBoxTensor,
+)
+from box_embeddings.initializations.tf_initializer import TFBoxInitializer
+from box_embeddings.initializations.tf_uniform_boxes import (
+    TFUniformBoxInitializer,
+)
+
+
+class TFBoxEmbedding(tf.keras.layers.Embedding):
+    """Embedding which returns boxes instead of vectors"""
+
+    def __init__(
+        self,
+        num_embeddings: int,
+        embedding_dim: int,
+        box_factory: TFBoxFactory = None,
+        box_initializer: TFBoxInitializer = None,
+        **kwargs: Any,
+    ) -> None:
+        box_factory = box_factory or TFBoxFactory("mindelta_from_vector")
+        super().__init__(
+            num_embeddings,
+            embedding_dim * box_factory.box_subclass.w2z_ratio,
+            **kwargs,
+        )
+        self.num_embeddings = num_embeddings
+        self.embedding_dim = self.output_dim
+        self.box_factory = box_factory
+
+        self.embeddings = self.add_weight(
+            shape=(self.input_dim, self.output_dim),
+            initializer=self.embeddings_initializer,
+            dtype=tf.float64,
+            name='embeddings',
+            regularizer=self.embeddings_regularizer,
+            constraint=self.embeddings_constraint,
+            experimental_autocast=False,
+        )
+
+        if box_initializer is None:
+            box_initializer = TFUniformBoxInitializer(
+                dimensions=embedding_dim,  # here dim is box dim
+                num_boxes=int(self.input_dim),  # type: ignore
+                box_type_factory=self.box_factory,
+            )
+        self.built = True
+        box_initializer(self.embeddings)
+
+    def call(self, inputs: tf.Tensor) -> TFBoxTensor:
+        emb = super().call(inputs)
```

Comment:
Don't use call explicitly, use the class as a callable. Like: `emb = super()(inputs)`

Reply:
I think this call is to the tf.keras.layers.Embedding's call function. To call that we would need to do a super().call(inputs) explicitly else we wont be able to inherit